src/components/inputBar.tsx

```diff
@@ -89,7 +114,144 @@ const InputBar = (props: InputBarProps) => {
       name: `${props.username},${props.userId}`,
       audio: "",
     };
+    if (props.dropZoneActive) {
+      setDisableInputs(true);
+      setIsRagLoading(true);
+
+      console.log("image dropped");
+      props.setInput("");
+      props.setDropzoneActive(false);
+
+      if (props.dropZoneImage && props.dropZoneImage.length > 0) {
+        const ID = nanoid();
+        const zodMessage: any = Schema.safeParse({
+          imageName: props.dropZoneImage[0].name,
+          imageType: props.dropZoneImage[0].type,
+          imageSize: props.dropZoneImage[0].size,
+          file: props.dropZoneImage[0],
+          value: props.value,
+          userId: props.userId,
+          orgId: props.orgId,
+          chatId: props.chatId,
+          message: [...props.messages, message],
+          id: ID,
+        });
+        const imageExtension = props.dropZoneImage[0].name.substring(
+          props.dropZoneImage[0].name.lastIndexOf(".") + 1,
+        );
+
+        // console.log("zodmessage", zodMessage);
+        // console.log("dropzone", props.dropZoneActive);
+        if (zodMessage.success) {
+          const message: Message = {
+            id: ID,
+            role: "user",
+            content: props.value,
+            name: `${props.username},${props.userId}`,
+          };
+          const file = props.dropZoneImage[0];
+          const zodMSG = JSON.stringify(zodMessage);
+          const formData = new FormData();
+          formData.append("zodMessage", zodMSG);
+          formData.append("file", file);
+          const response = await fetch("/api/imageInput", {
+            method: "POST",
+            body: formData,
+          });
+          const id = ID;
+          if (response) {
+            console.log("responce", response);
+            let assistantMsg = "";
+            const reader = response.body?.getReader();
+            console.log("reader", reader);
+            const decoder = new TextDecoder();
+            let charsReceived = 0;
+            let content = "";
+            reader
+              ?.read()
+              .then(function processText({ done, value }) {
+                if (done) {
+                  setDisableInputs(false);
+                  setIsRagLoading(false);
+                  console.log("Stream complete");
+                  fetch(`/api/updatedb/${props.chatId}`, {
+                    method: "POST",
+                    body: JSON.stringify({
+                      messages: [
+                        ...props.messages,
+                        awsImageMessage,
+                        message,
+                        {
+                          ...assistantMessage,
+                          content: content,
+                        },
+                      ],
+                      orgId: props.orgId,
+                      usreId: props.userId,
+                    }),
+                  });
+                  return;
```

Comment:
can not this new messages be saved to db in onCompletion callback of LangchainStream ?